### PR TITLE
Relax username requirements a little

### PR DESCRIPTION
**Pull request by ewels**
Minor changes to username requirements, needs agreement in meeting:

* Allow `.` and `-` in usernames
* Relax length requirements to 3-30 characters instead of 8-20

- [x] Tests passing
- [x] Black formatting
- [ ] Migrations for any changes to the database schema
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

**Comment by ewels** [comment]
Not sure why the tests are failing? Seems unrelated to the PR changes I think..

**Comment by inaod568** [comment]
> Not sure why the tests are failing? Seems unrelated to the PR changes I think..

Yep, it's the db encryption, have commented out the volumes in the db part of the docker-compose for now and then it works

File: dds_web/utils.py [hunk 1]
```diff
@@ -177,16 +177,7 @@ def _email_taken(form, field):
...
-    return string_contains_only(input=input, pattern=pattern)
-
-
-def string_contains_only(input, pattern):
-    """Check if string only contains specific characters."""
-    if re.search(pattern, input):
-        return True
-
-    return False
+    return re.search(r"^[a-zA-Z0-9_\.-]+$", input)
```
Comment (ewels) [inline dds_web/utils.py:180]:
```suggestion
    return False if re.search(r"^[a-zA-Z0-9_\.-]+$", input) == None else True
```

**Comment by inaod568** [comment]
Now you should be able to merge dev and then the tests should pass

**Comment by ewels** [comment]
I don't have permissions to merge :)

**Comment by inaod568** [comment]
I meant merge `dev` into this branch.

**Comment by inaod568** [comment]
So that the docker-compose change is in this too and the tests pass

**Comment by inaod568** [comment]
@ewels Could you also add an entry in the changelog?

**Review by inaod568 — COMMENTED** [review]
Just tested: Do we want to allow usernames containing all or almost all special characters?

**Review by inaod568 — APPROVED** [review]

**Pull request merged**